speclang_modern.tex

```diff
@@ -2099,6 +2099,40 @@ \subsection{Non-terminating functions}
 to always terminate, which is the expected behavior of most
 functions.
 
+The \lstinline|terminates| clause can also be specified in a statement contract
+with the same meaning. When such a clause is not specified, it inherits the
+one associated to the innermost block containing it (recursively, until a
+block with a \lstinline|terminates| clause is found or the body of the function
+is reached, and then the \lstinline|terminates| clause of the function applies).
+When a statement \lstinline|S| is specified with a \lstinline|terminates| clause
+\lstinline|ST|:
+
+\begin{itemize}
+  \item the termination clause \lstinline|ET| of the innermost block containing
+        it must imply \lstinline|ST|,
+  \item statements of \lstinline|S| must terminate if \lstinline|ST| holds, but
+        the termination clause \lstinline|ET| of any block containing
+        \lstinline|S| is ignored.
```

Comment:
`S` is not mentioned  in the next listing

Reply:
Removed